### ISSUE TITLE
Larger wheel and sdist size

### ISSUE THREAD

**Issue by jtpio** [jupyterlite/jupyterlite#1422]
## Description

The latest `0.4.0` pre-releases seem to be larger in size compared to the `0.3.0` releases:

## Reproduce

![image](https://github.com/jupyterlite/jupyterlite/assets/591645/cd47535c-7f81-421a-a21e-077fc623d892)

![image](https://github.com/jupyterlite/jupyterlite/assets/591645/5eeed5a6-0355-4ded-bbd9-48e9af495fdd)

![image](https://github.com/jupyterlite/jupyterlite/assets/591645/1082d633-426d-42b1-8190-e4887f3ec5f4)

![image](https://github.com/jupyterlite/jupyterlite/assets/591645/384ff4d6-f116-4df2-9daa-b66fee0c2a32)

## Expected behavior

We should check if this is expected or not.

## Context

A few new packages were added in:
- https://github.com/jupyterlite/jupyterlite/pull/1376
- https://github.com/jupyterlite/jupyterlite/pull/1397

**Comment by bollwyvl** [comment]
Yes. Without some analysis (e.g. `webpack-bundle-analyzer`), it can be very hard to tell what _exactly_ is the issue, but there is potentially some fat that can be cut. However, I still feel the most important metric remains the time-to-first pixels for users: ultimately the wheel size doesn't matter a whole lot for something (the wheel) that gets downloaded orders of magnitude fewer times than what's inside it (the core bundle that blocks the page).

As to the actual content: the biggest thing, by far, is the sourcemaps. We _could_ move them into a separate distribution which layered on top of `build` ... but having to redeploy to get sourcemaps would make it much harder to debug.

**Comment by bollwyvl** [comment]
(`diffoscope` is also available in the full linux dev environment and could show some details)

**Comment by jtpio** [comment]
For reference the `jupyterlab` assets also seem to have become larger over time:

![image](https://github.com/jupyterlite/jupyterlite/assets/591645/d0719129-edd5-4438-9a8f-662fa86d24a9)

![image](https://github.com/jupyterlite/jupyterlite/assets/591645/377c1b7c-3f38-409c-a69a-8a98036bf7f1)

**Comment by bollwyvl** [comment]
Without digging into `webpack-bundle-analyzer`, #1423 did some `resolutions` and `dedupe` (why oh why `npx yarn-berry-deduplicate`?), getting us back to actually _smaller_ that the last `0.3.0` dists...

**Comment by bollwyvl** [comment]
As we are no longer regressing on total package size, we _could_ close this out :tada:.

However, to avoid future issues like this one, we could:
- add a self-contained `webpack-bundle-analyzer` report as part of CI
  - this could be copied in during the docs build locally/on RTD
    - but would require someone to actually _look_ at it
- set some high-water values for our built `.tgz`, `.tar.gz`, and `.whl` during CI
  - fail a PR that increases artifact sizes by more than `x%` compared to the last `main` build
